src/cljs/nr/utils.cljs

```diff
@@ -185,25 +195,16 @@
       (map (fn [[k v]] [(regex-of k) v]))
       (sort-by (comp count str first) >))))
 
-(defn padded-interleave [pad & seqs]
-  "Interleave sequences of uneven lengths by padding out the shorter ones"
-  (let [lazy-padded-seqs (map #(concat % (repeat pad)) seqs)
-        num-seqs (count lazy-padded-seqs)
-        max-len (reduce max (map count seqs))]
-    (take (* max-len num-seqs) (apply interleave lazy-padded-seqs))))
-
 (defn replace-in-element [element [regex replacement]]
   "Given a string element, split that string on `regex`, then replace the
   matches removed by split with `replacement`. The replacement is performed by
-  first counting the number of matches, then interleaving that many
-  `replacement`s into the context. `padded-interleave` allows us to interleave
-  sequences of varying lengths by padding out the shorter of the two sequences,
-  in this case with empty strings."
+  interleaving `replacement`s into the context and dropping the last one as
+  interleave always weaves in excess"
   (if (string? element)
-    (let [context (split element regex)
-          match-count (count (re-seq regex element))
-          replacements (repeat match-count replacement)]
-      (->> (padded-interleave "" context replacements)
+    (let [context (.split element regex)
```

Comment:
any reason for `.split` instead of clojure's `split`?

Reply:
Yes! When I was testing, it was quite a bit faster when splitting with other strings rather than a regex (although it works with both). I suspect this is because the browser is better placed to optimise this. This is why in card-patterns-impl I just returned the string of the card rather than the regex. Also, clojure's split has a weird edge case on a full match where is simply returns an empty array, which makes the logic slightly more complex.

Reply:
Cool, that’s a great detail